terraform/modules/app-ecs-albs/main.tf

```diff
@@ -106,20 +106,26 @@ resource "aws_acm_certificate" "prometheus_cert" {
 }
 
 resource "aws_route53_record" "prometheus_cert_validation" {
-  # Count matches the domain_name plus each `subject_alternative_domain`
-  count = 1 + local.prom_records_count
-
-  name       = aws_acm_certificate.prometheus_cert.domain_validation_options[count.index]["resource_record_name"]
-  type       = aws_acm_certificate.prometheus_cert.domain_validation_options[count.index]["resource_record_type"]
-  zone_id    = var.zone_id
-  records    = [aws_acm_certificate.prometheus_cert.domain_validation_options[count.index]["resource_record_value"]]
-  ttl        = 60
+  for_each = {
```

Comment:
I'm not entirely convinced this will match the `1 +` logic so I'd appreciate someone familiar with this to check this carefully.